src/openApi/openApiWrapper.ts

```diff
@@ -55,6 +54,12 @@ export default class OpenApiWrapper {
             user: this.userPrincipal,
             query: request.multiValueQueryStringParameters,
           });
+
+          newrelic.addCustomAttributes({
+            canonicalId: this.userPrincipal,
+            correlationId,
+          });
+
```

Comment:
How does this work in a local environment when you don't have `newrelic` configured? I guess one can try it with `yarn link`

I think we will need to make the newrelic calls and import optional.

Reply:
Perhaps it can be even a configuration flag in the openApiWrapper constructor that allows enabling/disabling NR instrumentation since not everyone using this will want NR?

Reply:
Hmm good point(s). At this point I'm thinking we could even add some generic callbacks here and pass those via the constructor. So that the callers would need to require the newrelic package and do the webpack config exception at the same time.

Something like:
```
new OpenApiWrapper(
  logger,
  {
     onRequest: (..all the props in "this") => {...},
     onResponse...,
     onError...
  }
)
```
What do you think?

Reply:
Yeah, allowing to extend the callbacks is a good generic feature. I had a person from WMD asking about that feature before. 

That said,  I think having optional NR instrumentation in the shared lib is also useful because it will make it easy to have a unified format/instrumentation level across services. Also, not everyone uses webpack, for example, with esbuild the externals might not be needed.